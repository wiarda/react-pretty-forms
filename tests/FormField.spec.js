/* global describe test expect */
import React from 'react';
import renderer from 'react-test-renderer';
import { FormField } from '../index';

describe('Field component', () => {
  test('Matches snapshot', () => {
    const component = renderer.create(<FormField name="email" />).toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Renders required props', () => {
    const component = renderer
      .create(<FormField name="test" label="This is a test field" />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Renders required field', () => {
    const component = renderer
      .create(<FormField name="test" label="This is a test field" required />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Renders without a validation message', () => {
    const component = renderer
      .create(
        <FormField name="test" label="This is a test field" validationMessage={false} required />,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Accepts custom styles', () => {
    const component = renderer
      .create(
        <FormField name="test" label="This is a test field" className="test-class" validationMessage={false} required />,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
