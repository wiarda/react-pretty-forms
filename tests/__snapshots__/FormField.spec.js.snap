// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field component Accepts custom styles 1`] = `
<div
  className="pretty-row"
  style={Object {}}
>
  <label
    className="pretty-label"
    data-active={false}
    data-validity={true}
    htmlFor="test"
  >
    <span
      className="pretty-label-text"
      data-active={false}
    >
      This is a test field
    </span>
    <input
      className="test-class"
      data-validity={true}
      name="test"
      onBlur={[Function]}
      onFocus={[Function]}
      required={true}
      type="text"
    />
  </label>
  <div
    className="pretty-validation"
    data-validity={true}
  >
    
  </div>
</div>
`;

exports[`Field component Matches snapshot 1`] = `
<div
  className="pretty-row"
  style={Object {}}
>
  <input
    className="pretty-input"
    data-validity={true}
    name="email"
    onBlur={[Function]}
    onFocus={[Function]}
    required={false}
    type="text"
  />
  ;
  <div
    className="pretty-validation"
    data-validity={true}
  >
    Please enter a valid email address
  </div>
</div>
`;

exports[`Field component Renders required field 1`] = `
<div
  className="pretty-row"
  style={Object {}}
>
  <label
    className="pretty-label"
    data-active={false}
    data-validity={true}
    htmlFor="test"
  >
    <span
      className="pretty-label-text"
      data-active={false}
    >
      This is a test field
    </span>
    <input
      className="pretty-input"
      data-validity={true}
      name="test"
      onBlur={[Function]}
      onFocus={[Function]}
      required={true}
      type="text"
    />
  </label>
  <div
    className="pretty-validation"
    data-validity={true}
  >
    Please fill in this required field
  </div>
</div>
`;

exports[`Field component Renders required props 1`] = `
<div
  className="pretty-row"
  style={Object {}}
>
  <label
    className="pretty-label"
    data-active={false}
    data-validity={true}
    htmlFor="test"
  >
    <span
      className="pretty-label-text"
      data-active={false}
    >
      This is a test field
    </span>
    <input
      className="pretty-input"
      data-validity={true}
      name="test"
      onBlur={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
  </label>
  <div
    className="pretty-validation"
    data-validity={true}
  >
    
  </div>
</div>
`;

exports[`Field component Renders without a validation message 1`] = `
<div
  className="pretty-row"
  style={Object {}}
>
  <label
    className="pretty-label"
    data-active={false}
    data-validity={true}
    htmlFor="test"
  >
    <span
      className="pretty-label-text"
      data-active={false}
    >
      This is a test field
    </span>
    <input
      className="pretty-input"
      data-validity={true}
      name="test"
      onBlur={[Function]}
      onFocus={[Function]}
      required={true}
      type="text"
    />
  </label>
  <div
    className="pretty-validation"
    data-validity={true}
  >
    
  </div>
</div>
`;
